src/core/reference/src/runtime/opt_kernel/reshape.cpp

```diff
@@ -41,58 +42,107 @@ void reshape_in1(const char* in,
     }
 }
 
+static size_t get_threshold() {
+    // TODO: find a better way, not hardcoded value
+    return (1 << 9) * parallel_get_num_threads();
+}
+
+static inline void copy_element(char* out, const char* in, size_t elem_size) {
+#define CASE(type)                                                          \
+    case sizeof(type):                                                      \
+        *reinterpret_cast<type*>(out) = *reinterpret_cast<const type*>(in); \
+        break;
+
+    switch (elem_size) {
+        CASE(int32_t)
```

Comment:
maybe it's better to add a minor comment here, that float, double and so on types are also covered with this switch-case and this is expected.